src/main/java/seedu/address/logic/commands/tag/EditCommand.java

```diff
@@ -0,0 +1,155 @@
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Tag> contactTagList = model.getAddressBook().getContactTagList();
+        List<Tag> saleTagList = model.getAddressBook().getSaleTagList();
+
+        if (targetIndex.getOneBased() > contactTagList.size() + saleTagList.size() || targetIndex.getOneBased() < 0) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TAG_DISPLAYED_INDEX);
+        }
+
+        if (targetIndex.getOneBased() > contactTagList.size()) {
+            Tag tagToEdit = saleTagList.get(targetIndex.getZeroBased());
+            Tag editedTag = createEditedTag(tagToEdit, editTagDescriptor);
+            if (!tagToEdit.isSameTag(editedTag) && model.hasSaleTag(editedTag)) {
+                throw new CommandException(MESSAGE_DUPLICATE_TAG);
+            }
+
+            model.editContactTag(tagToEdit, editedTag);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+            return new CommandResult(String.format(MESSAGE_EDIT_TAG_SUCCESS, tagToEdit, editedTag));
+        }
+
+        Tag tagToEdit = contactTagList.get(targetIndex.getZeroBased());
+        Tag editedTag = createEditedTag(tagToEdit, editTagDescriptor);
+
+        if (!tagToEdit.isSameTag(editedTag) && model.hasContactTag(editedTag)) {
+            throw new CommandException(MESSAGE_DUPLICATE_TAG);
+        }
+
+        model.editContactTag(tagToEdit, editedTag);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        return new CommandResult(String.format(MESSAGE_EDIT_TAG_SUCCESS, tagToEdit, editedTag));
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code editTagDescriptor}.
+     */
+    private static Tag createEditedTag(Tag tagToEdit,
+                                       EditTagDescriptor editTagDescriptor) {
+        assert tagToEdit != null;
+
+        String updatedTagName = editTagDescriptor.getTagName().orElse(tagToEdit.getTagName());
+
+        return new Tag(updatedTagName);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditCommand)) {
+            return false;
+        }
+
+        // state check
+        EditCommand e = (EditCommand) other;
+        return targetIndex.equals(e.targetIndex)
+                && editTagDescriptor.equals(e.editTagDescriptor);
+    }
+
+    /**
+     * Stores the details to edit the person with. Each non-empty field value will replace the
+     * corresponding field value of the person.
+     */
+    public static class EditTagDescriptor {
+        private String tagName;
+
+        public EditTagDescriptor() {}
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
```

Comment:
Is this comment still relevant? If I'm not wrong it refers to the `tags` stored in the persons/sales right?